File: test-suite/src/test/java/io/micronaut/chatbots/basecamp/AboutCommandHandler.java
package io.micronaut.chatbots.basecamp;

import io.micronaut.chatbots.basecamp.api.Query;
import io.micronaut.chatbots.basecamp.core.BasecampBotConfiguration;
import io.micronaut.chatbots.basecamp.core.BasecampHandler;
import io.micronaut.chatbots.core.CommandResponse;
import io.micronaut.chatbots.core.TextResourceLoader;
import io.micronaut.context.annotation.Requires;
import io.micronaut.core.annotation.NonNull;
import io.micronaut.core.annotation.Nullable;
import jakarta.inject.Singleton;
import jakarta.validation.constraints.NotNull;

import java.util.Optional;

@Requires(property = "spec.name", value = "basecamp")
// tag::class[]
@Singleton
class AboutCommandHandler implements BasecampHandler {

    public static final String ABOUT = "about";
    private final TextResourceLoader textResourceLoader;

    AboutCommandHandler(TextResourceLoader textResourceLoader) {
        this.textResourceLoader = textResourceLoader;
    }

    @Override
    public boolean canHandle(@Nullable BasecampBotConfiguration bot, @NonNull @NotNull Query input) {
        return input.getCommand().equalsIgnoreCase("/" + ABOUT);
    }

    @Override
    public @NonNull Optional<String> handle(@Nullable BasecampBotConfiguration bot, @NonNull @NotNull Query input) {
        return textResourceLoader
            .composeCommandResponse(ABOUT)
            .map(CommandResponse::text);
    }
}
// end::class[]

File: test-suite-python/src/test/python/micronaut/chatbots/docs/basecamp/AboutCommandHandler.py
from jakarta.inject import Singleton
from java.util import Optional
from micronaut.chatbots.basecamp.api import Query
from micronaut.chatbots.basecamp.core import BasecampBotConfiguration, BasecampHandler
from micronaut.chatbots.core import TextResourceLoader
from micronaut.context.annotation import Requires


@Requires(property="spec.name", value="basecamp")
# tag::class[]
@Singleton
class AboutCommandHandler(BasecampHandler):

    ABOUT: str = "about"

    def __init__(self, text_resource_loader: TextResourceLoader):
        self.text_resource_loader = text_resource_loader

    def canHandle(self, bot: BasecampBotConfiguration | None, input: Query) -> bool:
        return input.getCommand().lower() == "/" + AboutCommandHandler.ABOUT

    def handle(self, bot: BasecampBotConfiguration | None, input: Query) -> Optional[str]:
        return (self.text_resource_loader
                .composeCommandResponse(AboutCommandHandler.ABOUT)
                .map(lambda response: response.text()))
# end::class[]
